samples/matter/common/src/bt_nus_service.cpp

```diff
@@ -146,8 +146,30 @@ bool NUSService::RegisterCommand(const char *const name, size_t length, CommandC
 
 void NUSService::DispatchCommand(const char *const data, uint16_t len)
 {
+	static constexpr char LF = '\n';
+	static constexpr char CR = '\r';
+	static const char *CRLF = "\r\n";
+
+	const size_t rawLenEol = len - 1;
+	const size_t rawLenWindowsEol = len - 2;
+
+	const char eol = *(data + rawLenEol);
+	const char *windowsEol = data + rawLenWindowsEol;
+
+	auto isNoEolCommand = [data, len](const Command &c) {
+		return strncmp(data, c.command, len) == 0 && len == strlen(c.command);
+	};
+	auto isEolCommand = [data, len, rawLenEol, eol](const Command &c) {
+		return strncmp(data, c.command, rawLenEol) == 0 && rawLenEol == strlen(c.command) &&
+		       (eol == LF || eol == CR);
+	};
+	auto isWindowsEolCommand = [data, len, rawLenWindowsEol, windowsEol](const Command &c) {
```

Comment:
If this returns true then `isEolCommand` also has to, so this is actually redundant?

Reply:
Considering we have different `len` for each of these cases I honestly don't get it :) can you explain?

Reply:
Ok, my bad, in the `isEolCommand` case we would compare the command string to `input\r` so it wouldn't match. Nevermind :)